ci/unit_tests/functions_delete_package/main_test.py

```diff
@@ -0,0 +1,179 @@
+"""
+Copyright 2019 IBM Corporation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json, os, pytest, requests, shutil, unittest, uuid, zipfile
+
+import functions_delete_package
+from ...test_utils import BaseTestCaseCapture
+from urllib.parse import quote
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+class TestMain(BaseTestCaseCapture):
+
+    dataBasePath = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'main_data')
+    packageBase = "Package-for-WAW-CI-"
+
+    def setup_class(cls):
+        BaseTestCaseCapture.checkEnvironmentVariables(['CLOUD_FUNCTIONS_USERNAME', 'CLOUD_FUNCTIONS_PASSWORD',
+                                                       'CLOUD_FUNCTIONS_NAMESPACE'])
+        cls.username = os.environ['CLOUD_FUNCTIONS_USERNAME']
+        cls.password = os.environ['CLOUD_FUNCTIONS_PASSWORD']
+        cls.apikey = cls.username + ':' + cls.password
+        cls.cloudFunctionsUrl = os.environ.get('CLOUD_FUNCTIONS_URL',
+                                               'https://us-south.functions.cloud.ibm.com/api/v1/namespaces')
+        cls.namespace = os.environ['CLOUD_FUNCTIONS_NAMESPACE']
+        cls.urlNamespace = quote(cls.namespace)
+        cls.actionsUrl = cls.cloudFunctionsUrl + '/' + cls.urlNamespace + '/actions/'
+        cls.functionFilePath = os.path.join(cls.dataBasePath, 'examplePython.py')
+
+    def callfunc(self, *args, **kwargs):
+        functions_delete_package.main(*args, **kwargs)
+
+    def setup_method(self):
+        self.package = self.packageBase + str(uuid.uuid4())
+
+    def _uploadPackage(self):
```

Comment:
We discussed this with @ota-sandr and it yould be better to use our functions_deploy script for more readability and lower probability of bugs ;)

Reply:
Done.